### ISSUE TITLE
Colour issues in GeSHi code blocks

### ISSUE THREAD

**Issue by selfthinker** [splitbrain/dokuwiki#3453]
The [GeSHi](https://www.dokuwiki.org/devel:geshi) code blocks have some colour issues.

* [ ] When words in code snippets are linked, those links are only differentiated by colour. Although those links are difficult to notice for everyone, so more a usability than an accessibility issue.
That fails [1.4.1 Use of Color](https://www.w3.org/TR/WCAG21/#use-of-color) and has a low priority.

* [ ] Lots of the code snippets have an insufficient colour contrast. That will make them harder to read for people with vision impairments.
That fails [1.4.3 Contrast (Minimum)](https://www.w3.org/TR/WCAG21/#contrast-minimum) and has a medium priority.

I went through all the possible colours from [geshi.less](https://github.com/splitbrain/dokuwiki/blob/master/lib/styles/geshi.less) and checked which ones fail the minimum contrast requirements.

| class                                  | colour    | contrast against `#FBFAF9` | WCAG (AA normal text) |
|----------------------------------------|-----------|----------------------------|-----------------------|
| .co0                                   | `#666666` |                        5.5 | ✅ pass                |
| .co4                                   | `#cc0000` |                       5.64 | ✅ pass                |
| .es5                                   | `#006699` |                       5.99 | ✅ pass                |
| .es6                                   | `#009933` |                       3.59 | ❌ fail                |
| .kw2                                   |` #000000` |                      20.14 | ✅ pass                |
| .kw5                                   | `#008000` |                       4.92 | ✅ pass                |
| .kw6                                   | `#ff0088` |                       3.59 | ❌ fail                |
| .me0                                   |` #004000` |                      11.62 | ✅ pass                |
| .nu0                                   | `#cc66cc` |                       3.17 | ❌ fail                |
| .re0                                   | `#0000ff` |                       8.24 | ✅ pass                |
| .re3                                   | `#ff3333` |                       3.48 | ❌ fail                |
| .re4                                   | `#009999` |                       3.34 | ❌ fail                |
| .re5                                   | `#660033` |                      12.49 | ✅ pass                |
| .re7                                   | `#991111` |                       8.24 | ✅ pass                |
| .re8                                   | `#00b000` |                       2.79 | ❌ fail                |
| .sc-2                                  | `#404040` |                       9.94 | ✅ pass                |
| .sy3                                   | `#000040` |                      18.75 | ✅ pass                |
| .br0, .sy0                             | `#66cc66` |                       1.93 | ❌ fail                |
| .co1, .coMULTI, .sc-1                  | `#808080` |                       3.78 | ❌ fail                |
| .co2, .sy1                             | `#339933` |                        3.5 | ❌ fail                |
| .co3, .sy4                             | `#008080` |                       4.57 | ✅ pass                |
| .es0, .es1, .esHARD                    | `#000099` |                      13.79 | ✅ pass                |
| .es2, .es3, .es4                       | `#660099` |                       9.94 | ✅ pass                |
| .kw1, .kw8                             | `#b1b100` |                       2.19 | ❌ fail                |
| .kw10, .kw11, .kw12, .kw9              | `#003399` |                      10.41 | ✅ pass                |
| .kw13, .kw14, .kw15, .kw16, .me1, .me2 | `#006600` |                       6.94 | ✅ pass                |
| .kw3, .kw7, .sy2                       | `#000066` |                       16.9 | ✅ pass                |
| .kw4, .re2                             | `#993333` |                          7 | ✅ pass                |
| .re1, .st0, .st_h                      | `#ff0000` |                       3.83 | ❌ fail                |

That needs some fiddling with colours. Maybe other projects could be an inspiration, like https://github.com/ericwbailey/a11y-syntax-highlighting or https://github.com/mpchadwick/pygments-high-contrast-stylesheets.
